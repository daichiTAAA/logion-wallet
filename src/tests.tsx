import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
