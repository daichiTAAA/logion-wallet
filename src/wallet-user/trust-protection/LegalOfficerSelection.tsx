import React, {useState} from "react";
import CreateProtectionRequestForm from "./CreateProtectionRequestForm";
import ConfirmProtectionRequest from "./ConfirmProtectionRequest";
import './TrustProtection.css';

export enum State {
    START,
    REQUEST_PROTECTION_DONE
}

export interface Props {
    initialState?: State
}

export default function LegalOfficerSelection(props: Props) {

    const [state, setState] = useState(props.initialState !== undefined ? props.initialState : State.START)

    return (
        <div className="TrustProtection">
            {state === State.START && (
                <CreateProtectionRequestForm
                    onSubmit={() => setState(State.REQUEST_PROTECTION_DONE)}/>
            )}
            {state === State.REQUEST_PROTECTION_DONE && (
                <ConfirmProtectionRequest/>
            )}
        </div>
    );

}
