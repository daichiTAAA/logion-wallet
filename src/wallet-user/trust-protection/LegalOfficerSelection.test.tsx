import React from "react";
import {shallowRender} from "../../tests";
import LegalOfficerSelection, {State} from "./LegalOfficerSelection";

test("START: Show form", () => {
    const tree = shallowRender(<LegalOfficerSelection/>)
    expect(tree).toMatchSnapshot();
});

test("REQUEST_PROTECTION_DONE: Show message", () => {
    const tree = shallowRender(<LegalOfficerSelection initialState={State.REQUEST_PROTECTION_DONE}/>)
    expect(tree).toMatchSnapshot();
});
