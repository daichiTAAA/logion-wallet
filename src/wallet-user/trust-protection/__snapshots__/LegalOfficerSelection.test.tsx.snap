// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REQUEST_PROTECTION_DONE: Show message 1`] = `
<div
  className="TrustProtection"
>
  <ConfirmProtectionRequest />
</div>
`;

exports[`START: Show form 1`] = `
<div
  className="TrustProtection"
>
  <CreateProtectionRequestForm
    onSubmit={[Function]}
  />
</div>
`;
